t/lib/WTSI/NPG/Data/ConsentWithdrawnTest.pm

```diff
@@ -0,0 +1,148 @@
+package WTSI::NPG::Data::ConsentWithdrawnTest;
+
+use strict;
+use warnings;
+use English qw(-no_match_vars);
+use Test::More;
+use Test::Exception;
+use File::Temp qw( tempdir );
+use Log::Log4perl;
+use Test::MockObject;
+
+use WTSI::NPG::iRODS;
+
+use base 'WTSI::NPG::HTS::Test';
+
+Log::Log4perl::init('./etc/log4perl_tests.conf');
+
+my $pid          = $PID;
+my $test_counter = 0;
+my $collection;
+
+sub setup_test : Test(setup) {
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  $collection =
+    $irods->add_collection("RunPublisherTest.$pid.$test_counter");
+  $test_counter++;
+}
+
+sub teardown_test : Test(teardown) {
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  $irods->remove_collection($collection);
+}
+
+sub object_and_query_creation : Test(6) {
+
+  use_ok 'WTSI::NPG::Data::ConsentWithdrawn';
+
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  my $cw = WTSI::NPG::Data::ConsentWithdrawn->new(irods => $irods);
+  isa_ok($cw, 'WTSI::NPG::Data::ConsentWithdrawn');
+
+  throws_ok { WTSI::NPG::Data::ConsentWithdrawn->new() }
+    qr/Attribute \(irods\) is required/,
+    'constructor should have irods attr. defined';
+
+  my $query = q{"%s/%s" "select COLL_NAME, DATA_NAME where } .
+    q{META_DATA_ATTR_NAME = 'sample_consent_withdrawn' and } .
+    q{META_DATA_ATTR_VALUE = '1' and DATA_NAME not like '%header.bam%'"};
+
+  is($cw->_iquery, q{iquest --no-page } . $query,
+    'default iquery, no zone');
+
+  $cw = WTSI::NPG::Data::ConsentWithdrawn->new(irods => $irods);
+  is($cw->_iquery, q{iquest --no-page } . $query,
+    'iquery with zone set');
+
+  $cw = WTSI::NPG::Data::ConsentWithdrawn->new(irods      => $irods,
+                                               zone       => 'seq',
+                                               collection => '/some/other');
+  is($cw->_iquery, q{iquest --no-page -z seq } . substr($query, 0, -1) .q{ and COLL_NAME like '/some/other%'"},
+    'iquery with both zone and collection set');
+}
+
+sub permissions : Test(21) {
+
+  my $mock = Test::MockObject->new();
+  $mock->fake_new( 'MIME::Lite' );
+  $mock->set_true('send');
+
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+
+  my @data_files = qw /1.bam 2.bam 3.bam 5.cram 6.cram 7.cram 9.cram/;
+  my @files = @data_files;
+  push @files, qw /1.bai 3.bai 4.bai 5.cram.crai 7.cram.crai 9.cram.crai
+                   10.fastq 11.cram.crai/;
+  my $tdir = tempdir( CLEANUP => 1 );
+  foreach my $file (@files) {
+    my $source = "$tdir/$file";
+    `touch $source`;
+    my $target = "$collection/$file";
+    $irods->add_object($source, $target);
+  }
```

Comment:
I'd prefer to see test setup in in the setup method rather than in the test method - for me, it makes it easier to disable teardown for debugging (all the setup/teardown in those methods) and keeps the test method focussed tests.

Reply:
moved in 3723c48